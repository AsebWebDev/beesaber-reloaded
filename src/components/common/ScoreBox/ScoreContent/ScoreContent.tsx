/* eslint-disable no-console */
import { MDBTable, MDBTableBody, MDBTableHead } from 'mdb-react-ui-kit';
import styled from 'styled-components';

import tokens from '@/tokens';

import HighScoresForOneSong from './HighScoresForOneSong/HighScoresForOneSong';

import type { Scores } from '@/sharedTypes';

const TableHead = styled(MDBTableHead)`
  color: ${tokens.color.yellow.main};
`;

const TableBody = styled(MDBTableBody)`
  color: ${tokens.color.white.main};
`;

type Props = {
  scores: Scores;
};

const ScoreContent = ({ scores }: Props): JSX.Element => (
  <MDBTable>
    <TableHead>
      <tr>
        <th scope="col">Rank</th>
        <th scope="col">Song</th>
        <th scope="col">Score</th>
        <th scope="col">Time</th>
      </tr>
    </TableHead>
    <TableBody>
      {scores.map((score) => (
        <HighScoresForOneSong highscore={score} key={score.scoreId} />
      ))}
    </TableBody>
  </MDBTable>
);

export default ScoreContent;
