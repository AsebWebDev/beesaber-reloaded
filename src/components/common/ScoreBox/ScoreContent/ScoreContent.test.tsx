import { render } from '@testing-library/react';

import exampleScores from '@/testing/testData/exampleScores';

import ScoreContent from './ScoreContent';

describe('components/common/ScoreContent/', () => {
  beforeAll(() => {
    jest.useFakeTimers('modern');
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  it('should match snapshot', () => {
    const { container } = render(<ScoreContent scores={exampleScores} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
