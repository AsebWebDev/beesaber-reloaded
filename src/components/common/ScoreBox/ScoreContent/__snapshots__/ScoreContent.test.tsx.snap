// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/ScoreContent/ should match snapshot 1`] = `
.c4 {
  margin: 1px 2px 1px 2px;
  max-height: 20px;
  max-width: -webkit-fit-content;
  max-width: -moz-fit-content;
  max-width: fit-content;
  min-width: 50px;
}

.c2 {
  color: #00111f!important;
  min-width: 30px;
}

.c5 {
  max-width: 50px;
  max-height: 30px;
  border-radius: 20px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1rem;
}

.c0 {
  color: #ffeb00;
}

.c1 {
  color: rgba(255,255,255,1);
}

<table
  class="table"
>
  <thead
    class="c0"
  >
    <tr>
      <th
        scope="col"
      >
        Rank
      </th>
      <th
        scope="col"
      >
        Song
      </th>
      <th
        scope="col"
      >
        Score
      </th>
      <th
        scope="col"
      >
        Time
      </th>
    </tr>
  </thead>
  <tbody
    class="c1"
  >
    <tr
      class=""
      data-testid="highscore-for-one-song"
    >
      <td>
        <span
          class="badge bg-primary c2"
        >
          2
        </span>
      </td>
      <td>
        <div
          class="c3"
        >
          <span
            class="c4 badge badge-danger"
          >
            Expert+
          </span>
          <img
            alt="Cover of It has Begun"
            class="c5"
            src="https://new.scoresaber.com/api/static/covers/32C3A0E1FEE9FD589B795941B6D79AFE8061D30A.png"
          />
          <span
            class="badge bg-dark"
          >
            Starset
             - 
            It has Begun
            <span
              class="greyed-out"
            >
               by 
              Joshabi
            </span>
          </span>
        </div>
      </td>
      <td>
        <span
          class="badge bg-primary"
          id="badge-light"
        >
          1000
        </span>
      </td>
      <td
        class="time"
      >
        <b
          class="card-link"
        >
          in 3 months
        </b>
      </td>
    </tr>
    <tr
      class=""
      data-testid="highscore-for-one-song"
    >
      <td>
        <span
          class="badge bg-primary c2"
        >
          2
        </span>
      </td>
      <td>
        <div
          class="c3"
        >
          <span
            class="c4 badge badge-danger"
          >
            Expert+
          </span>
          <img
            alt="Cover of SongPlayedByHive"
            class="c5"
            src="https://new.scoresaber.com/api/static/covers/32C3A0E1FEE9FD589B795941B6D79AFE8061D30C.png"
          />
          <span>
            <span>
              <i
                class="far fa-handshake"
              />
            </span>
            <div
              class="also-played-by"
            >
              <table>
                <thead>
                  <tr>
                    <th>
                      Bee
                    </th>
                    <th>
                      Score
                    </th>
                    <th>
                      My Score
                    </th>
                  </tr>
                </thead>
                <tbody
                  id="bees"
                />
              </table>
            </div>
          </span>
          <span
            class="badge bg-dark"
          >
            SongPlayedByHive
             - 
            SongPlayedByHive
            <span
              class="greyed-out"
            >
               by 
              Joshabi
            </span>
          </span>
        </div>
      </td>
      <td>
        <span
          class="badge bg-primary"
          id="badge-light"
        >
          3000
        </span>
      </td>
      <td
        class="time"
      >
        <b
          class="card-link"
        >
          in 3 months
        </b>
      </td>
    </tr>
    <tr
      class="played-by-hive-won"
      data-testid="highscore-for-one-song"
    >
      <td>
        <span
          class="badge bg-primary c2"
        >
          2
        </span>
      </td>
      <td>
        <div
          class="c3"
        >
          <span
            class="c4 badge badge-danger"
          >
            Expert+
          </span>
          <img
            alt="Cover of Let us make a deal"
            class="c5"
            src="https://new.scoresaber.com/api/static/covers/32C3A0E1FEE9FD589B795941B6D79AFE8061D30B.png"
          />
          <span>
            <span>
              <i
                class="far fa-handshake"
              />
            </span>
            <div
              class="also-played-by"
            >
              <table>
                <thead>
                  <tr>
                    <th>
                      Bee
                    </th>
                    <th>
                      Score
                    </th>
                    <th>
                      My Score
                    </th>
                  </tr>
                </thead>
                <tbody
                  id="bees"
                />
              </table>
            </div>
          </span>
          <span
            class="badge bg-dark"
          >
            Stray from the Path
             - 
            Let us make a deal
            <span
              class="greyed-out"
            >
               by 
              Joshabi
            </span>
          </span>
        </div>
      </td>
      <td>
        <span
          class="badge bg-primary"
          id="badge-light"
        >
          2000
        </span>
      </td>
      <td
        class="time"
      >
        <b
          class="card-link"
        >
          in 3 months
        </b>
      </td>
    </tr>
  </tbody>
</table>
`;
